substrate/frame/scheduler/src/lib.rs

```diff
@@ -415,6 +415,29 @@ pub mod pallet {
 			Self::service_agendas(&mut weight_counter, now, u32::MAX);
 			weight_counter.consumed()
 		}
+
+		#[cfg(feature = "std")]
+		fn integrity_test() {
+			fn lookup_weight<T: Config>(s: u32) -> Weight {
```

Comment:
maybe add some comment why is this important and/or what does it ensure exactly and/or why do we need to calculate all the weights bellow.

Isn't this related somehow to `MarginalWeightInfo`? Can we reuse `MarginalWeightInfo` here?
Looks like it does something like:
```
T::WeightInfo::service_agendas_base() +
   T::WeightInfo::service_agenda_base(T::MaxScheduledPerBlock::get())
   <T::WeightInfo as MarginalWeightInfo>::service_task(...);
```

Reply:
Yes thanks, using that seems easier.